crates/settings/src/settings_file.rs

```diff
@@ -115,7 +115,8 @@ pub fn update_settings_file<T: Settings>(
         let new_text = cx.read_global(|store: &SettingsStore, _cx| {
             store.new_text_for_update::<T>(old_text, update)
         })?;
-        fs.atomic_write(paths::SETTINGS.clone(), new_text).await?;
+        let resolved_settings_path = fs.canonicalize(paths::SETTINGS.clone().as_path()).await?;
```

Comment:
Since we're touching this, let's do a few more improvements:
* add a context to every callsite changed, with the path printed, so that we ensure we get the information about it in the error, if something goes wrong.

This is not `std::fs` which is notorious for not printing those paths, but I'd stay on the safe side and still add those.

* avoid the extra `clone` in `.clone().as_path()`.

In the end, we will get something like

```rust
let resolved_settings_path = fs
    .canonicalize(paths::SETTINGS.as_path())
    .await
    .with_context(|| {
        format!(
            "Failed to canonicalize settings path {:?}",
            paths::SETTINGS.as_path()
        )
    })?;
fs.atomic_write(resolved_settings_path.clone(), new_text)
    .await
    .with_context(|| {
        format!(
            "Failed to write settings to file {:?}",
            resolved_settings_path
        )
    })?;
```

+ the `anyhow::Context` trait import.